x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/color_map_select.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, Fragment } from 'react';
+
+import { EuiSuperSelect, EuiSpacer } from '@elastic/eui';
+import { ColorStopsOrdinal } from './color_stops_ordinal';
+import { COLOR_MAP_TYPE } from '../../../../../../common/constants';
+import { ColorStopsCategorical } from './color_stops_categorical';
+
+const CUSTOM_COLOR_MAP = 'CUSTOM_COLOR_MAP';
+
+export class ColorMapSelect extends Component {
+  state = {
+    selected: null,
+  };
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.customColorMap === prevState.prevPropsCustomColorMap) {
+      return null;
+    }
+
+    return {
+      prevPropsCustomColorMap: nextProps.customColorMap, // reset tracker to latest value
+      customColorMap: nextProps.customColorMap, // reset customColorMap to latest value
+    };
+  }
+
+  _onColorMapSelect = selectedValue => {
```

Comment:
This method would read better without declaring `newProps` only to deconstruct a line latter

```
_onColorMapSelect = selectedValue => {
    const useCustomColorMap = selectedValue === CUSTOM_COLOR_MAP;
    this.props.onChange({
      color: useCustomColorMap ? null : selectedValue,
      useCustomColorMap,
      type: this.props.colorMapType,
    });
  };
```